### PR TITLE
task042 added

### PR DESCRIPTION
**Pull request by gun-pi**
task042

### CODE REVIEW

File: src/main/java/com/epam/university/java/core/task042/Task042.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     *     result is "10:15"
+     *     Example: schedule ["09:30-10:15", "12:20-19:00"], currentTime "13:00",
+     *     result is "not available today"
+     * </p>
+     *
+     * @param schedule list with a schedule for a given day
+     * @param currentTime current time
+     * @return availability string
+     */
+    String checkAvailability(List<String> schedule, String currentTime);
```
Comment (aabarmin) [inline src/main/java/com/epam/university/java/core/task042/Task042.java]: Isn't it better using `enum` instead of `String` here?

**Comment by aabarmin** [comment]
Great suggestion, thanks.

File: src/main/java/com/epam/university/java/core/task042/Task042.java (hunk 1 above)
Comment (aabarmin) [reply src/main/java/com/epam/university/java/core/task042/Task042.java]: Oh, I see it sometimes returns suggested time. Anyway, I assume, it's better having interface like `BookingResponse` with implementations like `BusyResponse`, `AvailableResponse` and `TimeProposalResponse`.

Comment (gun-pi) [reply src/main/java/com/epam/university/java/core/task042/Task042.java]: Thank you! All your recommendations are implemented in the last commit.

Comment (aabarmin) [reply src/main/java/com/epam/university/java/core/task042/Task042.java]: Looks great now

**Pull request merged**